zerver/views/streams.py

```diff
@@ -75,12 +77,69 @@
 from zerver.lib.typed_endpoint import ApiParamConfig, PathOnly, typed_endpoint
 from zerver.lib.typed_endpoint_validators import check_color, check_int_in_validator
 from zerver.lib.user_groups import access_user_group_for_setting
-from zerver.lib.users import access_user_by_email, access_user_by_id
+from zerver.lib.users import access_user_by_email, access_user_by_id, access_user_common
 from zerver.lib.utils import assert_is_not_none
 from zerver.models import NamedUserGroup, Realm, Stream, UserProfile
 from zerver.models.users import get_system_bot
 
 
+def bulk_principals_to_user_profiles(
+    agent: UserProfile, principals: list[str] | list[int]
```

Comment:
s/agent/acting_user/, and make it the second parameter; so `bulk_principlals_to_user_profiles(principals: ..., acting_user: UserProfile)`; that's our normal convention in the codebase for functions specifying the current user, and since we should be deleting the old function, there's no reason to copy its interface.

Reply:
> s/agent/acting_user/

Sorry what is this ?

Reply:
Nope, [TypeError: Subscripted generics cannot be used with class and instance checks]

Reply:
By simple deduction, if `principals` is only `str`s OR only `int`s (because if not, will raise an error before executing even executing anything the endpoint), then checking the first element only is more than enough.
[see my test in CZO](https://chat.zulip.org/#narrow/stream/49-development-help/topic/stream.20principals/near/1909001)